elastic/kibana
# [Dashboard] Filter out experimental visualizations when labs setting is disabled

Closes https://github.com/elastic/kibana/issues/132882

## Summary

Before this change, any visualization types that were in their `experimental` stage showed up in the visualization dropdown regardless of whether or not the advanced setting `visualize:enableLabs` was true or false. However, if `visualize:enableLabs` was false and users tried to create one of these experimental visualizations, they would be redirected to an error page:

![image](https://user-images.githubusercontent.com/8698078/178831362-a1479bba-5701-4e0f-96f3-ac0ab726866e.png)

Now, as part of getting each visualization type for the menu, all `experimental` visualizations will be filtered out if the `visualize:enableLabs` UI setting is false - this prevents users from creating `experimental` visualizations when they do not have the appropriate settings enabled:

![image](https://user-images.githubusercontent.com/8698078/178832250-fd172e20-deeb-40c9-8811-43823c46e2f2.png)


### How to Test
1. Disable the `visualize:enableLabs` setting
2. Navigate to a new or existing dashboard
3. Click on the "Select type" dropdown menu
4. You should not see the old deprecated input controls in the visualization menu - note that this is currently the only experimental visualization, so every other one should remain as usual.

### Flaky Test Runner

![image](https://user-images.githubusercontent.com/8698078/179009093-a6164a83-58d1-4d6e-a521-e8778ed7f32c.png)

### Checklist


### For maintainers
